src/planscape/boundary/management/commands/upload_boundary.py

```diff
@@ -0,0 +1,89 @@
+from pathlib import Path
+
+from config.boundary_config import BoundaryConfig
+from django.conf import settings
+from django.contrib.gis.utils.layermapping import LayerMapping
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models.signals import pre_save
+
+from boundary.models import Boundary, BoundaryDetails
+
+
+class Command(BaseCommand):
+    help = ('Uploads a boundary layer specified in the boundary configuration file '
+            'into the Planscape database.')
+
+    def add_arguments(self, parser):
+        parser.add_argument('data_directory', nargs='?', type=str,
+                            help='Location of the boundary data.')
+        parser.add_argument('--boundary', nargs='?', type=str,
+                            help=('Name of the boundary to upload; if missing, all '
+                                  'boundaries are uploaded.'))
+        parser.add_argument('--configuration_file', nargs='?', type=str,
+                            help='Path to the boundary.json file; default is BASE_DIR/config/boundary.json.')
+        parser.add_argument('--force',
+                            type=bool, default=False, action=argparse.BooleanOptionalAction,
+                            help='Overwrite the boundary layer if it exists.')
+        parser.add_argument('--verbose',
+                            type=bool, default=True, action=argparse.BooleanOptionalAction,
+                            help='Show the commands to be run but do not run them.')
+
+    def handle(self, *args, **options):
+        data_directory = options['data_directory']
+        if data_directory is None:
+            raise CommandError(
+                'Must specify the data directory as an argument.')
+        configuration_file = options['configuration_file']
+        if configuration_file is None:
+            configuration_file = os.path.join(
+                settings.BASE_DIR, 'config/boundary.json')
+        boundary_to_load = options['boundary']
+        force = options['force']
+        verbose = options['verbose']
+
+        def presave_callback_generator(fkey):
+            def cb(sender, instance, *args, **kwargs):
+                instance.boundary = fkey
+            return cb
+
+        # Build the configuration object.
+        config = BoundaryConfig(configuration_file)
+
+        # Read the shapefiles and add Boundary and BoundaryDetail objects.
+        for boundary in config.get_boundaries():
+            # Check if need to load just one boundary
+            boundary_name = boundary['boundary_name']
+            if boundary_to_load is not None and boundary_to_load != boundary_name:
+                continue
+
+            # Create the new top-level Boundary
+            print("Creating boundary " + boundary_name)
+            display_name = boundary.get('display_name', None)
+            region_name = boundary.get('region_name', None)
+            if region_name is not None and region_name == "none":
```

Comment:
is there a particular reason why we want to have the manual "none" entry vs leaving it empty?

Reply:
It's in the boundary.json file, and I kinda like it being called out (see below for an example).  I'm happy to change it if you think that's better---can you file a bug against me if so?

{
            "boundary_name": "task_force_regions",
            "display_name": "Wildfire Task Force Regions",
            "region_name": "none",
            "filepath": "boundary/RFFC_Regions_2022/tier1_rffc_areas_jan2022.shp",
            "source_srs": 3310,
            "geometry_type": "MULTIPOLYGON",
            "shapefile_field_mapping": {
                "shape_name": "RFFC_tier1"
            }
        },